Source/WTF/wtf/JSONValues.cpp

```diff
@@ -646,36 +725,30 @@ void Value::writeJSON(StringBuilder& output) const
             output.append(m_value.number);
         break;
     }
-    default:
+    case Type::Object:
+    case Type::Array:
         ASSERT_NOT_REACHED();
     }
 }
 
 size_t Value::memoryCost() const
+{
+    return visitDerived([&](auto& value) {
+        return value.memoryCostImpl();
+    });
+}
+
+size_t Value::memoryCostImpl() const
 {
     size_t memoryCost = sizeof(this);
     if (m_type == Type::String && m_value.string)
         memoryCost += m_value.string->sizeInBytes();
     return memoryCost;
 }
 
-ObjectBase::~ObjectBase()
-{
-}
-
-RefPtr<Object> ObjectBase::asObject()
-{
-    static_assert(sizeof(Object) == sizeof(ObjectBase), "cannot cast");
-    return static_cast<Object*>(this);
-}
-
-RefPtr<const Object> ObjectBase::asObject() const
-{
-    static_assert(sizeof(Object) == sizeof(ObjectBase), "cannot cast");
-    return static_cast<const Object*>(this);
-}
+ObjectBase::~ObjectBase() = default;
 
-size_t ObjectBase::memoryCost() const
+size_t ObjectBase::memoryCostImpl() const
 {
     size_t memoryCost = Value::memoryCost();
```

Comment:
This needs to be changed to `memoryCostImpl` to avoid infinite recursion.

Reply:
Fixed.